src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala

```diff
@@ -49,40 +50,44 @@ object MCSUtils {
       MCSNodeMap: mutable.HashMap[String, MCSNode],
       lats: Array[Double],
       lons: Array[Double],
-      tightestBox: Boolean): Unit = {
+      tightestBox: Boolean,
+      localDir: String = "/tmp",
+      hdfsDir: String = null): Unit = {
 
     val (srcMCSNode, dstMCSNode) = getMCSNodes(edge, MCSNodeMap)
     val (srcNodeID, srcNodeGrid) = extract_masked_data(srcMCSNode, lats, lons, tightestBox)
     val (dstNodeId, dstNodeGrid) = extract_masked_data(dstMCSNode, lats, lons, tightestBox)
-    MCSUtils.writeNodeToNetCDF("/tmp/", srcNodeID, srcNodeGrid, lats, lons)
-    MCSUtils.writeNodeToNetCDF("/tmp/", dstNodeId, dstNodeGrid, lats, lons)
+    MCSUtils.writeNodeToNetCDF(localDir, srcNodeID, srcNodeGrid, lats, lons, hdfsDir)
+    MCSUtils.writeNodeToNetCDF(localDir, dstNodeId, dstNodeGrid, lats, lons, hdfsDir)
   }
 
   /**
    * Writes the node to a netCDF file
    *
-   * @param directory The directory to write the file to
+   * @param localDir  The local directory to write the file to
    * @param fileName  String The full path and name of the netcdfFile.
    * @param varData   ucar.ma2.ArrayInt.D2  2D data to be written to the file.
    * @param lats      Array[Double] of the latitudes to be used
    * @param lons      Array[Double] of the longitudes to be used
+   * @param hdfsDir   The HDFS directory to write the file to
    */
   def writeNodeToNetCDF(
-      directory: String,
+      localDir: String,
       fileName: String,
       varData: ucar.ma2.ArrayInt.D2,
       lats: Array[Double],
-      lons: Array[Double]): Unit = {
-    val filepath = directory + fileName
+      lons: Array[Double],
+      hdfsDir: String): Unit = {
+    val currFile = localDir + System.getProperty("file.separator") + fileName
     try {
-      val fsplit = filepath.split("_")
+      val fsplit = currFile.split("_")
```

Comment:
Using regex is overkill considering that the filename here was created by the program in the format F#CE#_latmin_latmax_lonmin_lonmax.nc so there is no other character needed for this parsing.